src/lib/logger/recordFactory.ts

```diff
@@ -110,19 +111,23 @@ export default class RecordFactory {
 
         let indexedCustomFields: any = {};
         providedFields.forEach((value, key) => {
-            // Stringify, if necessary.
-            if ((typeof value) != "string") {
-                value = jsonStringifySafe(value);
-            }
-
             if ([CustomFieldsFormat.CloudLogging, CustomFieldsFormat.All, CustomFieldsFormat.Default].includes(customFieldsFormat)
```

Comment:
this whole block looks a little bit overcomplicated, and duplicates the stringify solution.
I am not sure if there is room to get this simplified, but maybe factoring out the stringify to a second method would at least get rid of the duplication

Reply:
The structure of the two sections is similar, but the logic is different. 
We could skip the type check though. I would expect `jsonStringifySafe()` to be a no-op if the input is already a string.